client/src/main/java/io/pravega/client/segment/impl/SegmentMetadataClientImpl.java

```diff
@@ -100,6 +103,32 @@ public void segmentAttributeUpdated(SegmentAttributeUpdated segmentAttributeUpda
             }
         }
         
+        @Override
+        public void segmentTruncated(SegmentTruncated segmentTruncated) {
+            log.debug("Received stream segment truncate result {}", segmentTruncated);
+            CompletableFuture<SegmentTruncated> future;
+            synchronized (lock) {
+                future = truncateSegmentRequests.remove(segmentTruncated.getRequestId());
+            }
+            if (future != null) {
```

Comment:
Is there a scenario where `future `can be null ? Is adding a `Preconditions.checkState ` an over kill ?

Reply:
Yes, it can be null if the future was already removed from the map due to a previous failure.